cpp-ch/local-engine/Functions/SparkFunctionTrim.cpp

```diff
@@ -104,127 +104,107 @@ namespace
         }
 
         ColumnPtr
-        executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & /*result_type*/, size_t input_rows_count) const override
+        executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr & result_type, size_t input_rows_count) const override
         {
-            const ColumnString * src_str_col = checkAndGetColumn<ColumnString>(arguments[0].column.get());
-            if (!src_str_col)
-                throw Exception(ErrorCodes::ILLEGAL_COLUMN, "First argument of function {} must be String", getName());
-
+            const ColumnString * src_col = checkAndGetColumn<ColumnString>(arguments[0].column.get());
+            const ColumnConst * src_const_col = checkAndGetColumnConst<ColumnString>(arguments[0].column.get());
+            const ColumnString * trim_col = checkAndGetColumn<ColumnString>(arguments[1].column.get());
+            const ColumnConst * trim_const_col = checkAndGetColumnConst<ColumnString>(arguments[1].column.get());
+
+            String src_const_str;
+            String trim_const_str;
+            if (src_const_col)
+                src_const_str = src_const_col->getValue<String>();
+            if (trim_const_col)
+                trim_const_str = trim_const_col->getValue<String>();
+            if (trim_const_col && trim_const_str.empty()) {
+                return arguments[0].column->cloneResized(input_rows_count);
+            }
 
-            if (const auto * trim_const_str_col = checkAndGetColumnConst<ColumnString>(arguments[1].column.get()))
+            if (src_const_col && trim_const_col)
```

Comment:
It will never happen if `useDefaultImplementationForConstants` is true. DO NOT NEED consider the situation where both argument are const column.

Reply:
A good suggestion. It seems that Spark will simplify the function result with constant params.